lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1918,6 +1927,83 @@ postTransactionFee ctx (ApiT wid) body = do
         minCoins <- NE.toList <$> liftIO (W.calcMinimumCoinValues @_ @k wrk outs)
         liftHandler $ mkApiFee Nothing minCoins <$> W.estimateFee runSelection
 
+constructTransaction
+    :: forall ctx s k n.
+        ( ctx ~ ApiLayer s k
+        , Bounded (Index (AddressIndexDerivationType k) 'AddressK)
+        , GenChange s
+        , HardDerivation k
+        , HasNetworkLayer IO ctx
+        , IsOwned s k
+        , Typeable n
+        , Typeable s
+        , WalletKey k
+        )
+    => ctx
+    -> ArgGenChange s
+    -> ApiT WalletId
+    -> ApiConstructTransactionData n
+    -> Handler (ApiConstructTransaction n)
+constructTransaction ctx genChange (ApiT wid) body = do
+    let toAddressAmount (ApiPaymentAddresses content) =
+            addressAmountToTxOut <$> content
+        toAddressAmount (ApiPaymentAll _) = undefined -- this will be tackled when migration is supported
```

Comment:
Let's make sure our app doesn't implode when the user finds an implemented feature.
Can we use HTTP 501 instead of `undefined` or `error`, wherever possible?

```suggestion
    let toAddressAmount (ApiPaymentAddresses content) = pure (addressAmountToTxOut <$> content)
         toAddressAmount (ApiPaymentAll _) = liftE ErrTemporarilyDisabled -- TODO: this will be tackled when migration is supported
```